src/main/java/org/folio/dew/batch/acquisitions/edifact/mapper/EdifactMapper.java

```diff
@@ -18,51 +19,46 @@
 import org.folio.dew.domain.dto.VendorEdiOrdersExportConfig;
 import org.folio.dew.domain.dto.acquisitions.edifact.EdiFileConfig;
 
-public class PurchaseOrdersToEdifactMapper {
-  private final CompositePOConverter compositePOConverter;
+public class EdifactMapper implements ExportResourceMapper {
+  private final CompOrderEdiConverter compOrderEdiConverter;
 
-  public PurchaseOrdersToEdifactMapper(CompositePOConverter compositePOConverter) {
-    this.compositePOConverter = compositePOConverter;
+  public EdifactMapper(CompOrderEdiConverter compOrderEdiConverter) {
+    this.compOrderEdiConverter = compOrderEdiConverter;
   }
 
-  public String convertOrdersToEdifact(List<CompositePurchaseOrder> compPOs, VendorEdiOrdersExportConfig ediExportConfig, String jobName) throws EDIStreamException {
-    return convertOrdersToEdifact(compPOs, List.of(), ediExportConfig, jobName);
-  }
-
-  public String convertOrdersToEdifact(List<CompositePurchaseOrder> compPOs, List<Piece> pieces, VendorEdiOrdersExportConfig ediExportConfig, String jobName) throws EDIStreamException {
+  public String convertForExport(List<CompositePurchaseOrder> compPOs, List<Piece> pieces, VendorEdiOrdersExportConfig ediExportConfig, String jobName) throws EDIStreamException {
     ByteArrayOutputStream stream = new ByteArrayOutputStream();
 
     EDIOutputFactory factory = EDIOutputFactory.newFactory();
     factory.setProperty(EDIOutputFactory.PRETTY_PRINT, true);
 
-    EDIStreamWriter writer = factory.createEDIStreamWriter(stream);
-
-    // Count of messages (one message per purchase order)
-    int messageCount = 0;
-    writer.startInterchange();
-    writeStartFile(writer);
-
-    EdiFileConfig ediFileConfig = new EdiFileConfig();
-    ediFileConfig.setFileId(StringUtils.right(jobName, 14));
-    ediFileConfig.setLibEdiCode(ediExportConfig.getEdiConfig().getLibEdiCode());
-    ediFileConfig.setLibEdiType(ediExportConfig.getEdiConfig().getLibEdiType().getValue().substring(0, 3));
-    ediFileConfig.setVendorEdiCode(ediExportConfig.getEdiConfig().getVendorEdiCode());
-    ediFileConfig.setVendorEdiType(ediExportConfig.getEdiConfig().getVendorEdiType().getValue().substring(0, 3));
-
-    writeInterchangeHeader(writer, ediFileConfig);
-
-    var poLineIdToPieces = pieces.stream().collect(groupingBy(Piece::getPoLineId));
-    // Purchase orders
-    for (CompositePurchaseOrder compPO : compPOs) {
-      compositePOConverter.convertPOtoEdifact(writer, compPO, poLineIdToPieces, ediFileConfig);
-      messageCount++;
+    try (EDIStreamWriter writer = factory.createEDIStreamWriter(stream)) {
```

Comment:
so previously it was some memory leak because stream was not closed properly?

Reply:
I see previously if exception is thrown - only in this case stream will not be closed. And it is imporved now

Reply:
This issue was reported by sonar and yes it could possible not be closed, now is resolved